### DISCUSSION TITLE
制品库应该长成什么样

### DISCUSSION THREAD

**Discussion by fanhousanbu** [go-atomci/atomci#151] [General]
1.6预期实现ci/cd分离，其中最重要的一部分就是制品库；

1. 制品库如何在流程中体现？
2. 制品库应该支持到什么程度？例如项目文档算不算？项目报告（测试报告，静态扫描结果等）算不算？
3. 制品库载体一般用什么呢？svn或者git或者oss，还是都支持？
4. 制品库上传是自己实现还是依赖jenkins呢？

欢迎讨论

**Comment by colynn** [comment]
我们可以先想下功能边界，我理解制品更多的是CI的产物，比如二进制文件，构建日志、镜像（可以独立对接存储至镜像仓库）， 

对于二进制文件， 制品库的载体是不是可以支持一个标准即可，比如 s3类型的存储

可以参考 [coding](https://coding.net/), [aliyun package](https://packages.aliyun.com/maven), [jfrog artifactory](https://jfrog.com/artifactory/)  获取更多的产品想法

**Reply by colynn** [reply]
https://github.com/argoproj/argo-workflows/blob/master/workflow/artifacts/artifacts.go
